pkg/kubernetes/kubernetes.go

```diff
@@ -307,14 +322,23 @@ func NewVirtualMachineWatcher(ctx context.Context, config *restclient.Config,
 	log := logger.GetLogger(ctx)
 
 	scheme := runtime.NewScheme()
+	log.Info("adding scheme for vm-operator versions v1alpha1, v1alpha2, v1alpha3")
 	err = vmoperatorv1alpha3.AddToScheme(scheme)
 	if err != nil {
 		log.Errorf("failed to add to scheme with err: %+v", err)
 	}
+	err = vmoperatorv1alpha2.AddToScheme(scheme)
+	if err != nil {
+		log.Errorf("failed to add to scheme with err: %+v", err)
+	}
+	err = vmoperatorv1alpha1.AddToScheme(scheme)
+	if err != nil {
+		log.Errorf("failed to add to scheme with err: %+v", err)
+	}
 
 	gvk := schema.GroupVersionKind{
-		Group:   vmoperatorv1alpha3.GroupVersion.Group,
-		Version: vmoperatorv1alpha3.GroupVersion.Version,
+		Group:   vmoperatorv1alpha1.GroupVersion.Group,
+		Version: vmoperatorv1alpha1.GroupVersion.Version,
```

Comment:
will this account for only v1alpha1 version of vm ?

Reply:
we will keep this as v1alpha1 for backward compatibility issue, if changed to v1alpha3 (vmoperatorv1alpha3.GroupVersion.Version) watcher is not able to not find machines with older versions.